# HG changeset patch
# User Boris Grozev <boris@jitsi.org>
# Date 1591119693 18000
#      Tue Jun 02 12:41:33 2020 -0500
# Node ID 95673d02b38f2241ea06422fedef2e5692a2befd
# Parent  2df11e00f7ed5db0e680433cd3754a9f77cf837b
Set the session.ip property before the session filters are initialized when using mod_websocket

diff -ur /usr/lib/prosody/core/sessionmanager.lua ./core/sessionmanager.lua
--- /usr/lib/prosody/core/sessionmanager.lua	2020-01-19 15:50:32.000000000 +0000
+++ ./core/sessionmanager.lua	2020-06-10 23:07:26.489092434 +0000
@@ -30,6 +30,7 @@

 local function new_session(conn)
	local session = { conn = conn, type = "c2s_unauthed", conntime = gettime() };
+	session.ip = conn.http_remote_ip or conn:ip();
	local filter = initialize_filters(session);
	local w = conn.write;
	session.send = function (t)
@@ -49,7 +50,6 @@
		end
		return true;
	end
-	session.ip = conn:ip();
	local conn_name = "c2s"..tostring(session):match("[a-f0-9]+$");
	session.log = logger.init(conn_name);

diff -r 2df11e00f7ed -r 95673d02b38f plugins/mod_websocket.lua
--- a/modules/mod_websocket.lua	Mon Jun 01 14:27:33 2020 -0500
+++ b/modules/mod_websocket.lua	Tue Jun 02 12:41:33 2020 -0500
@@ -245,6 +245,8 @@
 		return "";
 	end
 
+    -- Make the remote IP extracted from the HTTP headers available at session initialization time (before the session filters are applied)
+    conn.http_remote_ip = request.ip
 	conn:setlistener(c2s_listener);
 	c2s_listener.onconnect(conn);
 
diff -r 2df11e00f7ed -r 95673d02b38f util/session.lua
--- a/util/session.lua	Mon Jun 01 14:27:33 2020 -0500
+++ b/util/session.lua	Tue Jun 02 12:41:33 2020 -0500
@@ -23,7 +23,8 @@
 
 local function set_conn(session, conn)
 	session.conn = conn;
-	session.ip = conn:ip();
+    session.ip = conn.http_remote_ip or conn:ip();
+
 	return session;
 end
 
